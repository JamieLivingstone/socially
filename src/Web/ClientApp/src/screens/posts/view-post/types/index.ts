export type Post = {
  author: {
    name: string;
    username: string;
  };
  body: string;
  commentsCount: number;
  createdAt: string;
  likesCount: number;
  slug: string;
  tags: Array<{ name: string }>;
  title: string;
  updatedAt: string;
};
