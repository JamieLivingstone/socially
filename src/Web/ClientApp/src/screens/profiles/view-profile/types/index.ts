export type Profile = {
  name: string;
  username: string;
  commentsCount: number;
  followersCount: number;
  postsCount: number;
  following: boolean;
};
