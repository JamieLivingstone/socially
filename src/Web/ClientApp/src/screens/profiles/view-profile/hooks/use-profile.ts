import axios from 'axios';
import { useQuery } from 'react-query';

import { Profile } from '../types';

export function useProfile(username: string) {
  const { data } = useQuery(
    ['profile', username],
    async function fetchProfileRequest() {
      const { data } = await axios.get<Profile>(`/api/v1/profiles/${username}`);

      return data;
    },
    {
      suspense: true,
      useErrorBoundary: true,
    },
  );

  return {
    profile: data as Profile,
  };
}
