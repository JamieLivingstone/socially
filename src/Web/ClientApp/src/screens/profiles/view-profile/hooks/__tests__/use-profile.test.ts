import { renderHook } from '@testing-library/react-hooks';
import moxios from 'moxios';
import { expect } from 'vitest';

import Providers from '@providers';

import { Profile } from '../../types';
import { useProfile } from '../use-profile';

describe('useProfile', () => {
  test('fetches profile', async () => {
    const response: Profile = {
      commentsCount: 2,
      followersCount: 64,
      following: false,
      name: 'John Doe',
      postsCount: 29,
      username: 'john',
    };

    moxios.stubRequest('/api/v1/profiles/john', {
      status: 200,
      response,
    });

    const { result, waitForNextUpdate } = renderHook(() => useProfile('john'), { wrapper: Providers });

    await waitForNextUpdate();

    expect(result.current).toMatchSnapshot();
  });
});
