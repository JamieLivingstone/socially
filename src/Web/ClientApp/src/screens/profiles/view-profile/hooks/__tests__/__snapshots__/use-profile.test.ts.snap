// Vitest Snapshot v1

exports[`useProfile > fetches profile 1`] = `
{
  "profile": {
    "commentsCount": 2,
    "followersCount": 64,
    "postsCount": 29,
    "username": "john",
  },
}
`;
